src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509CertificateReader.cs

```diff
@@ -245,10 +245,7 @@ public RSA GetRSAPrivateKey()
                 return null;
             }
 
-            using (SafeRsaHandle rsaHandle = Interop.libcrypto.EVP_PKEY_get1_RSA(_privateKey))
```

Comment:
@bartonjs - are the two changes to this file OK? Doing it this way allowed me to not have to include a bunch of interop files in the X509Certificates.csproj. My naïve thinking is that this approach is a little different than existing code, but should be fine.  The current code calls EVP_PKEY_get1_* here and then passes in the IntPtr into the constructor, which then just duplicates the handle and then this code disposes of the first handle.  The new way is to pass in the privateKey into the constructor, which will call EVP_PKEY_get1_* and just hold on to that handle.  So this way seems simpler/more performant to me.

Thoughts?